src/common/components/Button.tsx

```diff
@@ -0,0 +1,90 @@
+import Link from 'next/link';
+
+import { IButtonGroup, IButton } from '../lib/interfaces';
+
+import { validateUrl } from '../utils/helpers/regex/url';
+import { ReactNode } from 'react';
+
+const options = ({ className }: { className: string | undefined }) => {
+  const disabledOptions = ` disabled:ring-opacity-50 disabled:bg-opacity-50 disabled:hover:bg-opacity-50 disabled:!cursor-not-allowed disabled:text-opacity-50 disabled:hover:text-opacity-40 disabled:hover:!scale-100 transform avoid-transform `;
+  if (className) {
+    return className + disabledOptions;
+  } else return disabledOptions;
+};
+
+const ButtonGroup = ({ className, children }: IButtonGroup) => {
+  return (
+    <div
+      className={`${
+        className ? className : ''
+      } box-border flex items-center flex-wrap`}
+    >
+      {children}
+    </div>
+  );
+};
+const Layout = ({
+  route,
+  children,
+}: {
+  route: string | undefined;
```

Comment:
`undefined` here is still required. would be better as `route?: string`

Reply:
Nice catch!